sktime/transformations/series/difference.py

```diff
@@ -0,0 +1,317 @@
+    }
+
+    def __init__(self, lags=1, remove_missing=True):
+        self.lags = lags
+        self.remove_missing = remove_missing
+        self._Z = None
+        self._cumulative_lags = None
+        self._prior_cum_lags = None
+        self._prior_lags = None
+        super(Differencer, self).__init__()
+
+    def _check_inverse_transform_index(self, Z):
+        """Check fitted series contains indices needed in inverse_transform."""
+        first_idx = Z.index.min()
+        orig_first_idx, orig_last_idx = self._Z.index.min(), self._Z.index.max()
+
+        is_contained_by_fitted_z = False
+        is_future = False
+
+        if first_idx < orig_first_idx:
+            msg = [
+                "Some indices of `Z` are prior to timeseries used in `fit`.",
+                "Reconstruction via `inverse_transform` is not possible.",
+            ]
+            raise ValueError(" ".join(msg))
+
+        elif Z.index.difference(self._Z.index).shape[0] == 0:
+            is_contained_by_fitted_z = True
+
+        elif first_idx > orig_last_idx:
+            is_future = True
+
+        pad_z_inv = self.remove_missing or is_future
+
+        cutoff = Z.index[0] if pad_z_inv else Z.index[self._cumulative_lags[-1]]
+        fh = ForecastingHorizon(
+            np.array([*range(-1, -(self._cumulative_lags[-1] + 1), -1)])
+        )
+
+        index = fh.to_absolute(cutoff).to_pandas()
+        index_diff = index.difference(self._Z.index)
+
+        if index_diff.shape[0] != 0:
+            msg = [
+                f"Inverse transform requires indices {index}",
+                "to have been stored in `fit()`,",
+                f"but the indices {index_diff} were not found.",
+            ]
+            raise ValueError(" ".join(msg))
+
+        return is_contained_by_fitted_z, pad_z_inv
+
+    def _fit(self, Z, X=None):
+        """Logic used by fit method on `Z`.
+
+        Parameters
+        ----------
+        Z : pd.Series or pd.DataFrame
+            A time series to apply the specified difference transformation on.
+
+        Returns
+        -------
+        self
+        """
+        self.lags = _check_lags(self.lags)
+        self._prior_lags = np.roll(self.lags, shift=1)
+        self._prior_lags[0] = 0
+        self._cumulative_lags = self.lags.cumsum()
+        self._prior_cum_lags = np.zeros_like(self._cumulative_lags)
+        self._prior_cum_lags[1:] = self._cumulative_lags[:-1]
+        self._Z = Z.copy()
+        return self
+
+    def _transform(self, Z, lags):
+        """Logic used by `transform` to apply transformation to `Z`.
+
+        Differences are applied at lags specified in `lags`.
+
+        Parameters
+        ----------
+        Z : pd.Series or pd.DataFrame
+            The timeseries to be differenced.
+
+        lags : np.ndarray
+            Lags to be used in applying differences.
+
+        Returns
+        -------
+        diff :
+            Differenced series.
+        """
+        diff = Z.copy()
```

Comment:
for readability, so that we don't have `diff.diff(...)`
```suggestion
        Zt = Z.copy()
```

Reply:
That makes sense to me. I've updated.